test/oTable.test.js

```diff
@@ -0,0 +1,428 @@
+		expect(document.getElementById("initiallyWrappedTable").parentNode.parentNode.classList.contains("test-wrapper")).to.be.false;
+	});
+
+	it("doesn't re-wrap tables", () => {
+		OTable.wrap(".test-container table", "test-wrapper");
+		expect(document.getElementById("initiallyUnwrappedTable").parentNode.parentNode.classList.contains("test-wrapper")).to.be.false;
+	});
+
+});
+
+describe("oTable API", () => {
+
+	it("is defined", () => {
+		expect(OTable).to.be.a('function');
+	});
+    
+    it('has a static init method', () => {
+		expect(OTable.init).to.be.a('function');
+	});
+    
+    it('has a destroy instance method', () => {
+		expect(OTable.prototype.destroy).to.be.a('function');
+	});
+    
+    it('has a removeEventListeners instance method', () => {
+		expect(OTable.prototype.removeEventListeners).to.be.a('function');
+	});
+    
+    it('has a sortRowsByColumn instance method', () => {
+		expect(OTable.prototype.sortRowsByColumn).to.be.a('function');
+	});
+    
+    it('has a dispatch instance method', () => {
+		expect(OTable.prototype.dispatch).to.be.a('function');
+	});
+
+});
+
+describe('An oTable instance', () => {
+    let oTableEl;
+    let testOTable;
+
+	beforeEach(() => {
+		sandbox.init();
+        sandbox.setContents(`
+            <table class="o-table" data-o-component="o-table">
+                <thead>
+                    <th>Cheese</th>
+                </thead>
+                <tbody>
+                    <tr>
+                        <td>cheddar</td>
+                    </tr>
+                    <tr>
+                        <td>stilton</td>
+                    </tr>
+                    <tr>
+                        <td>red leicester</td>
+                    </tr>
+                </tbody>
+            </table>
+        `);
+		oTableEl = document.querySelector('[data-o-component=o-table]');
+	});
+
+	afterEach(() => {
+		testOTable = undefined;
+		sandbox.reset();
+        oTableEl = undefined;
+	});
+
+	xit('is defined', () => {
+        testOTable = new OTable(oTableEl);
+		expect(testOTable).not.to.be.undefined;
+	});
+    
+    it('has the correct prototype', () => {
+        testOTable = new OTable(oTableEl);
+        expect(Object.getPrototypeOf(testOTable)).to.equal(OTable.prototype); 
+    });
+
+    it('sets a data attribute on the root element of the component to indicate the JS has executed', () => {
+        testOTable = new OTable(oTableEl);
+        expect(oTableEl.hasAttribute('data-o-table--js')).to.be.true;
+    });
+    
+    it('fires an "oTable.ready" event when the JS for the component has executed', done => {
+        testOTable = new OTable(oTableEl);
+        oTableEl.addEventListener('oTable.ready', function() {
+            done();
+        });
+    });
+    
+    it('sorts by ascending order first if not told otherwise', done => {
+        testOTable = new OTable(oTableEl);
+        // TODO - Add a click polyfill to polyfill-service
```

Comment:
I imagine something like this would work:

```js
HTMLElement.prototype.click = function() {
  var click = document.createEvent("MouseEvent");
  click.initMouseEvent("click", true, true, window,0, 0, 0, 0, 0, false, false, false, false, 0, null);
  this.dispatchEvent(click);
};
```